ibis/expr/datatypes/__init__.py

```diff
@@ -702,6 +790,16 @@ def __init__(
         self.value_type = dtype(value_type)
         self._pretty += f"<{self.value_type}>"
 
+    def _literal_value_hash_key(self, value):
+        return self, _tuplize(value)
+
+    def __fields_eq__(
+        self,
+        other: Set,
```

Comment:
can u put some of these in a super class?

Reply:
No, the point is to ensure that any subclass of `DataType` defines it and avoid any implicit behavior.

Reply:
got it makes sense